pkg/controller/batch/computeenvironment/setup.go

```diff
@@ -0,0 +1,553 @@
+		}
+	} */
+
+	if awsclients.StringValue(cr.Status.AtProvider.State) == string(svcapitypes.CEState_DISABLED) {
+		//if already or finally DISABLED, we are done here and the controller can delete the CE
+		return false, nil
+	}
+	//(ASK): wie können specs verändert werden? Oder stellen sie reinen Userinput dar, der nur bei Nil mit einem Standardwert z.B.
+	//in lateInitialize gesetzt werden dürfen? (der Nutzer steuert dann alles über .yaml?)
+	/* cr.Spec.ForProvider.DesiredState = awsclients.String(svcsdk.CEStateDisabled)
+	cr.Spec.DeepCopyInto(&cr.Spec)
+	fmt.Printf("cr.Spec.ForProvider.DesiredState: %v\n", awsclients.StringValue(cr.Spec.ForProvider.DesiredState)) */
+
+	/* in := GenerateUpdateComputeEnvironmentInput(cr).SetState(svcsdk.CEStateDisabled).SetComputeEnvironment(*obj.ComputeEnvironment)
+	in.SetComputeResources(nil) //bc some stats like e.g. Subnets or SGs can't be updated/modified
+	fmt.Printf("in: %v\n", in)
+
+	if _, err := e.client.UpdateComputeEnvironment(in); err != nil {
+		return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	} */ //leaving this here in case the way below makes problems again (not sure if it was a one time thing, misinterpreted or what)
+
+	//thought this could be faster for EC2 type Deletion, however unsure if the updating of vCPUs really ...
+	//CB: Test: only if time (as it needs to be tested for all types)
+	/* in := &svcsdk.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: awsclients.String(meta.GetExternalName(cr)),
+		State:              awsclients.String(svcsdk.CEStateDisabled)}
+	//MANAGED EC2 ... SPOT too?
+	if cr.Spec.ForProvider.Type == awsclients.String(svcsdk.CETypeManaged) && cr.Spec.ForProvider.ComputeResources.Type == awsclients.String(svcsdk.CRTypeEc2) {
+		in.ComputeResources.SetMinvCpus(0) //desired needs to be between min and max
+		in.ComputeResources.SetDesiredvCpus(0)
+	}
+	if _, err : = e.client.UpdateComputeEnvironment(in) {
+		return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	} */ //Alternative to make deletion maybe faster for EC2
+
+	_, err := e.client.UpdateComputeEnvironment(&svcsdk.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: awsclients.String(meta.GetExternalName(cr)),
+		State:              awsclients.String(svcsdk.CEStateDisabled)})
+	return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	//depending on the CE Setup (type, compute resources, status ...) Update-Call can take some while (-> status UPDATING) before getting to the Deletion...
+}
+
+func postDelete(_ context.Context, cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DeleteComputeEnvironmentOutput, err error) error {
+
+	//CB: possibly not needed when predelete solves all...
+	if err != nil {
+		if strings.Contains(err.Error(), "modified") {
+			// skip: Can't delete workspace in non-ACTIVE state. Current status is DELETING
+
+			fmt.Println("postDelete skipped err modified ", err)
```

Comment:
We should better skip deletion in `preDelete` if the resource is in non-active state. (I think we do this already?)